WeBankFinTech/FATE
# Fate 1.0 algorithm module update pull request

Fixes  ISSUE#313， 312，310，309，308， 307，306，305，304，303，256，227，194，193

Changes:

1. Update all algorithm modules running mechanism. Support federated modeling pipeline by FATE-Flow

2. Intermediate statistic result callback is avaliable and visualizable in FATE-Board for all algorithm modules.

3. Support sparse input-format in federated feature binning

4. Update evaluation metrics result

5. Update algorithm's parameter-define class

6. Add general cross-validation module

7. Update transfer variable generate mechanism

8. imputer updated

9. Add roc and gain for evaluation

10. Algorithm runtime conf definition and parameter-init method update

11. Update setting_conf mechanism for all algorithm modules

12. Model predict output adjust, adding more predict details

13. Algorithm parameter class decoupling , param-check and validation mechanism changes
